packages/upload-media/src/components/provider/with-registry-provider.tsx

```diff
@@ -0,0 +1,59 @@
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+import { useRegistry, createRegistry, RegistryProvider } from '@wordpress/data';
+import { createHigherOrderComponent } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import { storeConfig } from '../../store';
+import { STORE_NAME as mediaUploadStoreName } from '../../store/constants';
+
+type WPDataRegistry = ReturnType< typeof createRegistry >;
+
+function getSubRegistry(
+	subRegistries: WeakMap< WPDataRegistry, WPDataRegistry >,
+	registry: WPDataRegistry,
+	useSubRegistry: boolean
+) {
+	if ( ! useSubRegistry ) {
+		return registry;
+	}
+	let subRegistry = subRegistries.get( registry );
+	if ( ! subRegistry ) {
+		subRegistry = createRegistry( {}, registry );
```

Comment:
I'm not sure I'm following this. Why is it creating a new registry?

Reply:
See https://github.com/WordPress/gutenberg/pull/66290#discussion_r1810383124 and subsequent implementation in #66380